### PR TITLE
remove deprecated, update deps, refactor

### PR DESCRIPTION
**Pull request by whitecolor**
- removed deprecated components
- added `registerRootComponent` param to the driver which allows custom registration (i.e. Expo case)
- updated dependencies
- moved rn, react to peerDependencies
- added ES6 (uncommented) compilation
- added prettier with default config

**For discussion:**
1) Prettier probably should go with single quote, any other preferences?
2) Driver params now:
```ts
export declare function makeReactNativeDriver(appKey: string, { registerRootComponent, }?: {
    registerRootComponent?: (component: React.ComponentType) => any;
}):
```

maybe should be something like this, not sure:
```ts
export declare function makeReactNativeDriver(
  appKeyOrOptions:
    | string
    | {
        appKey?: string
        registerRootComponent?: (component: React.ComponentType) => any;
      }
)
```

@staltz

### CODE REVIEW

**Comment by staltz** [comment]
Hey @whitecolor, thanks for the PR. I'll add some review comments inline, I think this PR could be smaller and do less changes.

File: package.json [hunk 1]
```diff
@@ -14,38 +14,49 @@
...
   "main": "lib/cjs/index.js",
   "typings": "lib/cjs/index.d.ts",
-  "types": "lib/cjs/index.d.ts",
   "dependencies": {
-    "@cycle/react": "1.4.x",
-    "react": "16.4.x",
-    "react-native": ">=0.55.0",
-    "xstream": "11.x.x"
+    "@cycle/react": "2.x",
+    "@cycle/run": "5.x",
```
Comment (staltz) [inline package.json:24]: `cycle/run` is used in tests, not in `src`, so I would not put it in `dependencies`, it should be only in `devDependencies`.

File: package.json [hunk 2]
```diff
@@ -14,38 +14,49 @@
...
     "compile-cjs": "tsc --module commonjs --outDir ./lib/cjs",
-    "compile-es6": "echo 'TODO' : tsc --module es6 --outDir ./lib/es6",
-    "test": "TS_NODE_PROJECT=test/tsconfig.json mocha test/*.ts --require @huston007/react-native-mock/mock.js --require ts-node/register --recursive"
-  }
-}
\ No newline at end of file
+    "compile-es6": "tsc --module es6 --outDir ./lib/es6",
+    "test": "ts-node -P test/tsconfig.json --require @huston007/react-native-mock/mock.js node_modules/mocha/bin/mocha test/*.ts"
+  },
+  "prettier": {}
```
Comment (staltz) [inline package.json:61]: Could you use the same prettier config as other cyclejs repos in this org? E.g. https://github.com/cyclejs/cyclejs/blob/313762d9beb87e099bb36511875c42bd7e1ff980/package.json#L57

File: package.json [hunk 3]
```diff
@@ -14,38 +14,49 @@
     "react-native",
     "driver"
   ],
+  "files": [
+    "lib"
```
Comment (staltz) [inline package.json:18]: should also include `LICENSE` and `readme.md`

**Review by staltz — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (whitecolor) [reply package.json:18]: LICENCE and readme are included by default: https://docs.npmjs.com/files/package.json#files, that's why `files` is a more appropriate way to include than `.npmignore`, files also support negate patterns.

File: package.json (hunk 2 above)
Comment (whitecolor) [reply package.json]: done.

Comment (staltz) [reply package.json]: Also, could you apply the new styles with prettier? I think especially the test files should be reformatted.

File: package.json [hunk 4]
```diff
@@ -14,38 +14,51 @@
...
+  ],
   "main": "lib/cjs/index.js",
   "typings": "lib/cjs/index.d.ts",
-  "types": "lib/cjs/index.d.ts",
   "dependencies": {
-    "@cycle/react": "1.4.x",
-    "react": "16.4.x",
-    "react-native": ">=0.55.0",
-    "xstream": "11.x.x"
```
Comment (staltz) [inline package.json:24]: xstream is a dependency because it's required in `src/driver`

**Review by whitecolor — COMMENTED** [review]
> xstream is a dependency because it's required in src/driver

It is not a dependency of the runtime, only typings. Do you think it is still should be in deps?

**Comment by staltz** [comment]
That's what we usually do in other repos in this org, yes.

**Comment by staltz** [comment]
Looking good. I hope you don't mind if I squash-merge. I'll put out a new release too.

**Pull request merged**